### PR TITLE
Add Missing ActiveSupport::Rescuable to ActionCable::Channel

### PR DESCRIPTION
**Pull request by ghost**
### Summary

ActionController uses Rescuable but it appears that ActionCable::Channel does not. Was this intentional? It would be very useful to include this so that `rescue_from` could be used like this:

```
class Channel < ActionCable::Channel::Base
  rescue_from ActiveRecord::RecordInvalid, with: :show_errors

  protected
    def show_errors(exception)
      exception.record.new_record? ? ...
    end
end
```

### CODE REVIEW

**Comment by senny** [comment]
Bringing in @dhh as this concerns the API of Action Cable. Any thoughts on how errors should be handled in a channel?

**Comment by dhh** [comment]
👍 from me. I think it was just an oversight that we didn't add this in already.

File: actioncable/lib/action_cable/channel/base.rb [hunk 1]
```diff
@@ -259,6 +261,8 @@ def dispatch_action(action, data)
           else
             public_send action
           end
+          rescue Exception => exception
+            rescue_with_handler(exception) || raise
```
Comment (matthewd) [inline actioncable/lib/action_cable/channel/base.rb]: This indentation is off... and we should probably add a test for the new behaviour too.

**Comment by sirwolfgang** [comment]
Any reason this PR is stale? I just tried to `rescue_from` only to find out that it wasn't supported.

**Comment by kaspth** [comment]
@sirwolfgang it's probably just one of those things. Give @timthez a day or two to pick this up again, then have at it with a new PR. Make sure to credit @timthez via https://contributors.rubyonrails.org/faq#credit-several-people

**Comment by kaspth** [comment]
Fixed in #34686

**Pull request closed without merging**